src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -562,17 +599,16 @@ try
         loader.ApplyRuntimeInitialSettings();
     }
 
-    loader.LayerInboxOntoUser();
+    loader.MergeInboxIntoUserSettings();
     // Fragments might reference user profiles created by a generator.
-    // --> LayerFragmentsOntoUser must be called after LayerInboxOntoUser.
-    loader.LayerFragmentsOntoUser();
-    loader.DisableDeletedProfiles();
+    // --> FindFragmentsAndMergeIntoUserSettings must be called after MergeInboxIntoUserSettings.
+    loader.FindFragmentsAndMergeIntoUserSettings();
+    // DisableDeletedProfiles returns true whenever we encountered any new generated/dynamic profiles.
+    // Coincidentially this is also the time we should write the new settings.json
+    // to disk (so that it contains the new profiles for manual editing by the user).
+    mustWriteToDisk |= loader.DisableDeletedProfiles();
     loader.FinalizeLayering();
 
-    // NOTE: loader is being moved into CascadiaSettings() below.
-    // Extract anything of value before that.
-    const auto mustWriteToDisk = firstTimeSetup || loader.userProfilesChanged;

```

Comment:
Is `userProfilesChanged` still present? Who updates it now? Who uses it?

Reply:
`userProfilesChanged` is gone entirely now.